File: src/my_cpp_pkg/src/smartphone.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"


class Smartphone : public rclcpp::Node
{
public:
  Smartphone() : Node("smartphone")
  {
    subscriber_ = this->create_subscription<example_interfaces::msg::String>(
    "robot_news", 10, std::bind(&Smartphone::CallbackRobotNews, this, std::placeholders::_1)); 
    RCLCPP_INFO(this->get_logger(), "Smartphone is ready to receive news!"); 
  }

private:
  void CallbackRobotNews(const example_interfaces::msg::String::SharedPtr msg)
  {
    RCLCPP_INFO(this->get_logger(), "Received news: '%s'", msg->data.c_str());
  }

  rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr subscriber_; 

};

int main (int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<Smartphone>();
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}   

File: src/my_py_pkg/my_py_pkg/smartphone.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String

class Smartphone(Node):
    def __init__(self):
        super().__init__('smartphone')
        self.get_logger().info('Smartphone node is up and running!')
        self.subscriber_ = self.create_subscription( 
            String, 'robot_news', self.robot_news_callback, 10)
        self.subscriber_  # prevent unused variable warning
        self.get_logger().info('Smartphone node is subscribed to the "robot_news" topic.')

    def robot_news_callback(self, msg):
        self.get_logger().info('Received news: "%s"' % msg.data)

def main(args=None):
    rclpy.init(args=args)
    node = Smartphone()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()       

if __name__ == '__main__':
    main()